src/vsr/replica.zig

```diff
@@ -2205,24 +2207,14 @@ pub fn ReplicaType(
             self.repair();
         }
 
-        fn reference_message_and_receive_quorum_exactly_once(
-            self: *Self,
-            messages: *QuorumMessages,
-            message: *Message,
-            threshold: u32,
-        ) ?usize {
-            assert(threshold >= 1);
-            assert(threshold <= self.replica_count);
-
+        fn add_message_to_quorum(self: *Self, messages: *QuorumMessages, message: *Message) void {
             assert(messages.len == constants.replicas_max);
             assert(message.header.cluster == self.cluster);
             assert(message.header.replica < self.replica_count);
             assert(message.header.view == self.view);
             switch (message.header.command) {
                 .do_view_change => {
                     assert(!self.solo());
-                    if (self.replica_count == 2) assert(threshold == 2);
-
```

Comment:
while we are at it, might replace switch with assert?

Reply:
Actually, it seems like we in general have some dead code here, `if (message.header.command == .do_view_change) {` bellow is always true now, no? Probably makes sense to refactor separately then.

Reply:
:+1: I'll update it. This function used to handle all of the quorums, but all of the others were changed to simple bitsets (for less RAM use).